core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/SchedulerServer.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.scheduler
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.headers.BasicHttpCredentials
+import akka.http.scaladsl.server.Route
+import org.apache.openwhisk.common.{Logging, TransactionId}
+import org.apache.openwhisk.http.BasicRasService
+import org.apache.openwhisk.http.ErrorResponse.terminate
+import spray.json._
+
+import scala.concurrent.ExecutionContext
+
+/**
+ * Implements web server to handle certain REST API calls.
+ * Currently provides a health ping route, only.
+ */
+class SchedulerServer(scheduler: SchedulerCore, systemUsername: String, systemPassword: String)(
+  implicit val ec: ExecutionContext,
+  implicit val actorSystem: ActorSystem,
+  implicit val logger: Logging)
+    extends BasicRasService {
+
+  override def routes(implicit transid: TransactionId): Route = {
+    super.routes ~ extractCredentials {
+      case Some(BasicHttpCredentials(username, password)) if username == systemUsername && password == systemPassword =>
+        (path("disable") & post) {
```

Comment:
what's the point of having this disable route?

Reply:
That is for the graceful shutdown.
Once we disable a scheduler, it no longer receives any new requests.
After we confirm there is no activation in all queues, we can finally stop the scheduler.

Reply:
I see that's left as todo for now for the graceful shutdown. So in theory the scheduler could hang forever if the queues don't drain? Or is there going to be some hard kill after some timeout?

Reply:
Can we rename the route to `shutdown`? `disable` I associate more with temporarily turning off the resource but not stopping the entire service. not a big deal if you want to leave it as is

Reply:
Generally, even if a scheduler is disabled, existing containers still can fetch data from it.
But in case there are some issues in the system and activations are not properly handled in time, we just drop them with a `whiskError`.
The grace period for dropping activations is configurable and we are using 10s internally.

Reply:
I hope we can keep it as the scheduler is still running and working as expected after it is disabled.
(It just no longer receives any new messages.)
And I believe we can easily add an `enable` feature to re-enable the scheduler without restarting the component itself.

Reply:
Yes I didn't realize the scheduler service would still be running on the disabled node. Disabled makes sense in that case.

Reply:
This exchange is helpful to include as comments in the code.

Reply:
Should `scheduler.shutdown()` then also be called `disable` (or pause)?